vars/withBuildStatus.groovy

```diff
@@ -0,0 +1,30 @@
+/**
+ * body's execution control that propagates build statuses to github according to the result
+ */
+def call(String context, String repourl, String sha, Closure body) {
+  currentBuild.result = 'SUCCESS'
+  setBuildStatus("", "PENDING", context, repourl, sha)
+  try {
+    body.call()
+    setBuildStatus("", "SUCCESS", context, repourl, sha)
+  } catch (Throwable cause) {
+    setBuildStatus(cause.toString().take(140), "FAILURE", context, repourl, sha)
+    throw cause
+  }
+}
+
+/**
+ * https://wiki.jenkins.io/display/JENKINS/GitHub+Plugin
+ */
+def setBuildStatus(String message, String state, String context, String repourl, String sha) {
```

Comment:
Does this make the setBuildStatus() method unavailable for a direct use? This step deserves a standalone usage.
Why isn't BUILD_URL a parameter too?